pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -254,7 +256,17 @@ func (alloc *Action) allocateResourceForTasksWithTopology(tasks *util.PriorityQu
 				klog.ErrorS(nil, "HyperNode not exists.", "jobName", job.UID, "name", hyperNodeName, "tier", tier)
 				continue
 			}
-
+			LCAHyperNodeMap[hyperNodeName] = hyperNodeName
+			// The job still has remaining tasks to be scheduled, check whether the least common ancestor hypernode still meets the requirement of the highest allowed tier
+			if jobHyperNode != "" {
+				LCAHyperNode := ssn.HyperNodes.GetLCAHyperNode(hyperNodeName, jobHyperNode)
+				hyperNodeInfo := ssn.HyperNodes[LCAHyperNode]
+				if hyperNodeInfo.Tier()+1 > highestAllowedTier {
```

Comment:
Shouldn't add 1 here, for example, s0 and s1 both are tier1 hypernode and they are sibling hypernodes, and current jobHyperNode is s1. Then LCAHyperNode is s4, which are s0 and s1's parent, tier 2 > highestAllowedTier, there is no need to add 1